fuel-types/src/canonical.rs

```diff
@@ -430,23 +430,40 @@ impl<const N: usize, T: Deserialize> Deserialize for [T; N] {
         } else {
             // Spec doesn't say how to deserialize arrays with unaligned
             // primitives(as `u16`, `u32`, `usize`), so unpad them.
-            let mut uninit = <MaybeUninit<[T; N]>>::uninit();
+            let mut uninit: [MaybeUninit<T>; N] =
```

Comment:
May I suggest this pattern instead: https://github.com/FuelLabs/fuel-vm/blob/dfab064ad084115d4d2cdd0191c28d71abe8679d/fuel-compression/src/impls.rs#L128-L154

Reply:
Done in 741007fd1